core-build/gulp-core-build/src/logging.ts

```diff
@@ -272,20 +271,33 @@ function _writeTaskError(e: any): void {
 }
 
 function exitProcess(errorCode: number): void {
-
   if (!localCache.watchMode) {
     process.stdout.write('', () => {
       process.exit(errorCode);
     });
   }
 }
 
-function wireUpProcessErrorHandling(): void {
+function wireUpProcessErrorHandling(shouldWarningsFailBuild: boolean | undefined): void {
   if (!wiredUpErrorHandling) {
     wiredUpErrorHandling = true;
+
+    let wroteToStdErr: boolean = false;
+
+    if (shouldWarningsFailBuild) {
+      const oldStdErr: Function = process.stderr.write;
+      // tslint:disable-next-line:no-function-expression
+      process.stderr.write = function (text: string | Buffer): boolean {
+        if (!!text.toString()) {
+          wroteToStdErr = true;
+          return oldStdErr.apply(process.stderr, arguments);
+        }
+        return true;
```

Comment:
>return true; [](start = 8, length = 12)

Why don't we pass this through?

Reply:
It seems something somewhere is writing empty stuff to stderr. Not sure what. It needs more investigation.

---
In reply to: [162232129](https://github.com/Microsoft/web-build-tools/pull/495#discussion_r162232129) [](ancestors = 162232129)